library/src/main/java/com/datatorrent/lib/db/jdbc/JdbcPOJOInputOperator.java

```diff
@@ -175,8 +175,16 @@ protected void populateColumnDataTypes() throws SQLException
       }
 
       if (query != null) {
+        List<Integer> invalidFields = Lists.newArrayList();
         for (FieldInfo fieldInfo : fieldInfos) {
-          columnDataTypes.add(nameToType.get(fieldInfo.getColumnName()));
+          if (nameToType.get(fieldInfo.getColumnName()) == null) {
+            invalidFields.add(fieldInfos.indexOf(fieldInfo));
+          } else {
+            columnDataTypes.add(nameToType.get(fieldInfo.getColumnName()));
+          }
+        }
+        for (int invalidField : invalidFields) {
```

Comment:
@d9liang is there an advantage of removing the corresponding entry from fieldInfos? If not then simply ignore adding it to columnDataTypes and don't add additional code to remove from fieldInfos.